src/commands/register.ts

```diff
@@ -1,46 +1,66 @@
+import { CommandInteraction, Message } from "discord.js";
 import { Db } from "mongodb";
 
-async function subcmd_new(interaction, db: Db) {
-    const gameTag = interaction.options.getString("val_id");
-    const pronouns = interaction.options.getString("pronouns");
-    const result = await db.collection("members").insertOne({
-        gameTag,
-        pronouns,
-        discordId: interaction.user.id,
-    });
+import { MessageExecutable } from "./command";
 
-    if (!result) {
-        interaction.reply({
-        content:
-            "Ah, my eyes are down. I couldn't register you; check with an admin.",
-        ephemeral: true,
-        });
-    } else {
-        interaction.reply({
-        content:
-            "I know **exactly** who you are; you're registered! Feel free to update with /register update",
-        ephemeral: true,
+class NewSubcommand extends MessageExecutable<CommandInteraction> {
+    async execute() {
+        const gameTag = this.interaction.options.getString("val_id");
+        const pronouns = this.interaction.options.getString("pronouns");
+        const existing_user = await this.db.collection("members").find({
```

Comment:
I think you can use `findOne` here to avoid needing to collect to an array. Should theoretically never be a problem, but is less work for you in the code.

Reply:
Resolved in a1de6ca2a60d054fe4df9297f512daab2b54e81d.